lib/enhanced_writer.rb

```diff
@@ -0,0 +1,32 @@
+class EnhancedWriter
```

Comment:
`EnhancedWriter` is here to show you the way you add feature without decorator pattern is appending more and more methods to a simple class. As you do it, you add lot of context in its unit test.
The more context you forced to be in a simple class test, more you need to think about closing this class for alteration. The good way is extracting surrounded behavior to another class. Check the `SimpleWriter` and the others `Writers` variation classes below.